### PR TITLE
[IMPROVEMENT] Added text and description fields for Image Attachment

### LINKED ISSUE #1381: [IMPROVEMENT] The layout of the attachment messages needs to be improved
## Description
The image attachment's title is not properly displayed and the description is not displayed at all in the Rocket.Chat android app. I have attached a screenshot too. 

screenshot from 2018-06-15 19-39-25

## Devices and Versions

Your Rocket.Chat.Android version: 2.3.2 Build 2026
Your Rocket.Chat Server version: open.rocket.chat

Mobile device model and OS version: Oneplus 3 Oreo 8.0

## Steps to reproduce
Send an image attachment with desc and title from web and see it on the android app.

I would love to work on this issue. It would be really helpful if someone can also give some pointers. :)

EDIT:
This is the log of `Message` Object that the `translate` function is receiving in the file chatroom/uimodel/UiModelMapper.kt
```
Message(id=jJseXdiGGXK97e6pz, roomId=L73ECFoZSeDJsBMsJWMRzgjyHDBQYhJ7h7, message=, timestamp=1528990675967, sender=SimpleUser(id=L73ECFoZSeDJsBMsJ, username=RichMessageTestBot, name=RichMessageTestBot), updatedAt=1528991058827, editedAt=null, editedBy=null, senderAlias=null, avatar=null, type=null, groupable=false, parseUrls=false, urls=null, mentions=[], channels=[], attachments=[ImageAttachment(title=Lauri M, description=null, titleLink=https://www.basketball-reference.com/players/m/markkla01.html, titleLinkDownload=false, imageUrl=http://www.trbimg.com/img-5b04c449/turbine/ct-spt-bulls-lauri-markkanen-all-rookie-team-20180522, imageType=null, imageSize=null, imagePreview=null)], pinned=false, starred=null, reactions={:mouse:=[utkarsh.barsaiyan]}, role=null, isTemporary=null, unread=null)
```
The above object does not contain a `text` field but has a `description` field in the image attachment set to `null`.

But in the following log the `text` field can be seen:
```
06-18 09:38:36.391 3231-3314/chat.rocket.android.dev D/TimberLogger: SDK(https://dev.veranda.viasat.io): Received text message: {"msg":"changed","collection":"stream-notify-user","id":"id","fields":{"eventName":"DmZw72BAjmsZGQ5YR/rooms-changed","args":["updated",{"_id":"DmZw72BAjmsZGQ5YRL73ECFoZSeDJsBMsJ","t":"d","_updatedAt":{"$date":1529329117013},"lastMessage":{"rid":"DmZw72BAjmsZGQ5YRL73ECFoZSeDJsBMsJ","attachments":[{"title":"Lauri M","title_link":"https://www.basketball-reference.com/players/m/markkla01.html","text":"Should have been rookie of the year","image_url":"http://www.trbimg.com/img-5b04c449/turbine/ct-spt-bulls-lauri-markkanen-all-rookie-team-20180522","color":"#164FA5"}],"ts":{"$date":1529329117000},"u":{"_id":"L73ECFoZSeDJsBMsJ","username":"RichMessageTestBot","name":"RichMessageTestBot"},"msg":"","mentions":[],"channels":[],"_updatedAt":{"$date":1529329117002},"_id":"Krp5pAjNDjxWyDrNh","sandstormSessionId":null}}]}}, channel: RendezvousChannel@f28520{EmptyQueue}
```
So, the text field gets lost between the JSON fetched from the server and the Image Attachment that is displayed. And there is some ambiguity between text and description fields. The description field is not mentioned in the docs of the image attachment. If anyone can provide any information to solve the issue, it would be of great help.

### PR DESCRIPTION
**Pull request by ubarsaiyan**
@RocketChat/android

Closes #1381 

In collaboration with pull [#189](https://github.com/RocketChat/Rocket.Chat.Kotlin.SDK/pull/189) of Rocket.Chat.Kotlin.SDK

### CODE REVIEW

File: app/src/main/java/chat/rocket/android/chatroom/uimodel/UiModelMapper.kt [hunk 1]
```diff
@@ -338,6 +340,17 @@ class UiModelMapper @Inject constructor(
         }
     }
 
+    private fun attachmentText(attachment: FileAttachment): String? {
+        return with(attachment) {
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/chatroom/uimodel/UiModelMapper.kt]: `with()` here is useless. Could you remove that, please?

File: app/src/main/java/chat/rocket/android/chatroom/uimodel/UiModelMapper.kt [hunk 2]
```diff
@@ -338,6 +340,17 @@ class UiModelMapper @Inject constructor(
...
+            return@with text
+        }
+    }
+    private fun attachmentDescription(attachment: FileAttachment): String? {
+        return with(attachment) {
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/chatroom/uimodel/UiModelMapper.kt]: `with()` here is useless. Could you remove that, please?

**Review by philipbrito — CHANGES_REQUESTED** [review]
You need only to "fix" two things and there we go!

**Comment by philipbrito** [comment]
@ubarsaiyan Could you fix the CI? This is being unable to generate the apk.

**Review by philipbrito — APPROVED** [review]
Thanks for your first PR here @ubarsaiyan 
Welcome :)

**Comment by ubarsaiyan** [comment]
Thank you 😊

**Pull request merged**